src/main/groovy/transform/Undefined.java

```diff
@@ -28,6 +28,10 @@
     private Undefined() {}
     public static final String STRING = "<DummyUndefinedMarkerString-DoNotUse>";
     public static final class CLASS {}
+    public static final class UNDEFINED_EXCEPTION extends RuntimeException {
```

Comment:
I think it would be more in line with the other Undefined fields if the 'UNDEFINED_' prefix were dropped from the field name e.g. 'EXCEPTION_CLASS' or something to that effect.